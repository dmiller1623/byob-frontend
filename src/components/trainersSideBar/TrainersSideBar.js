import React, { Component } from 'react';
import './trainersSideBar.css';
import Trainers from '../../images/Trainers.png';
import Delete from '../../images/delete.svg';
import PropTypes from 'prop-types';


class TrainersSideBar extends Component {
  constructor() {
    super();
  }

  render() {
    const { trainersTeams, trainers, toggleAddTrainer, deleteTrainer, deletePokemonTeam } = this.props;
    return (
      <div className='side-bar'>
        <img className='trainer-header' alt='trainer title' src={Trainers} />
        <button onClick={toggleAddTrainer}>Add Trainer</button>
        <div>
          {trainers.map((trainer, index) => {
            const foundTeams = trainersTeams.filter((team) => {
              return trainer.id === team.trainer_id;
            });
            return <div key={index}>
              <div className='name-header'>
                <h2>Name: <span>{trainer.name}</span></h2>
                <h2>Level: <span>{trainer.level}</span></h2>
                <img className='delete-buttons' alt='delete button' src={Delete} onClick={() => deleteTrainer(trainer.id)}/>
              </div>
              {foundTeams.map((pokemon, index) => {
                return <div key={index} className='team-display'>
                  <img className='sidebar-images' alt='pokemon' src={pokemon.pokemon_one}/>
                  <img className='sidebar-images' alt='pokemon' src={pokemon.pokemon_two}/>
                  <img className='sidebar-images' alt='pokemon' src={pokemon.pokemon_three}/>
                  <img className='sidebar-images' alt='pokemon' src={pokemon.pokemon_four}/>
                  <img className='sidebar-images' alt='pokemon' src={pokemon.pokemon_five}/>
                  <img className='delete-buttons' src={Delete} alt='delete button' onClick={() => deletePokemonTeam(pokemon.id)}/>
                </div>;
              })}
            </div>;
          })}
        </div>
      </div>
    );
  }
}

TrainersSideBar.propTypes = {
  trainersTeams: PropTypes.array,
  trainers: PropTypes.array,
  toggleAddTrainer: PropTypes.func,
  deleteTrainer: PropTypes.func,
  deletePokemonTeam: PropTypes.func
};

export default TrainersSideBar;