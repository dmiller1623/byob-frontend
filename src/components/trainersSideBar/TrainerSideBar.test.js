import React from 'react';
import { shallow } from 'enzyme'; 
import TrainersSideBar from './trainersSideBar.js';

describe('TrainerSideBar', () => {
  let wrapper;
  let trainersTeamsMock = []
  let trainersMock =[]
  let deleteTrainerMock
  let toggleAddTrainerMock
  let deletePokemonTeamMock

  it('should match the snapshot', () => {
    wrapper = shallow(<TrainersSideBar trainers={trainersMock} trainersTeam={trainersTeamsMock}/>);
    expect(wrapper).toMatchSnapshot();
  });
});  