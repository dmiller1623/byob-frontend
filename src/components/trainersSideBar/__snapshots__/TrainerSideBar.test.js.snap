// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainerSideBar should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrainersSideBar
    trainers={Array []}
    trainersTeam={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="trainer title"
          className="trainer-header"
          src="Trainers.png"
        />,
        <button>
          Add Trainer
        </button>,
        <div />,
      ],
      "className": "side-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "trainer title",
          "className": "trainer-header",
          "src": "Trainers.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Trainer",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Add Trainer",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="trainer title"
            className="trainer-header"
            src="Trainers.png"
          />,
          <button>
            Add Trainer
          </button>,
          <div />,
        ],
        "className": "side-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "trainer title",
            "className": "trainer-header",
            "src": "Trainers.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Trainer",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Add Trainer",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
